kitchen-sink-example.yaml

```diff
@@ -26,7 +61,45 @@ logger_provider:
     attribute_count_limit: 128
 
 # Configure meter provider.
-meter_provider: {}
+meter_provider:
+  metric_readers:
+    - type: prometheus
```

Comment:
Should this be `pull`?

Was the issue of readers/exporters being so tightly bound in their definitions discussed in the otel-schema repo or one of the meetings? I feel like in the spec there was discussion about potentially having users just configure exporters and the Reader be sort of figured out from that...

Defining the exporter and the Reader being "figured out" from that is I guess sort of what is being done if `type` can be `prometheus`, instead of type being `pull` and `prometheus` being an exporter defined under that. But then shouldn't the same be done for `otlp`?

Reply:
By saying that the `type` is prometheus, we're more firmly asserting that for the purposes of file configuration, prometheus is a specific pull based reader, not an exporter.

> I feel like in the spec there was discussion about potentially having users just configure exporters and the Reader be sort of figured out from that...

Not defining the reader means we need to have somewhere to define the reader properties, like the interval for periodic metric reader. Feels wrong to put this on the exporter configuration, since it would result in metric exporter configuration being quite different from trace and log exporter configuration.

Reply:
By using a type of `prometheus` it is already sort of merging the two (reader and exporter) but instead of defining the exporter the user defines the reader. 

Maybe that makes the most sense and there would be `type: otlp`?

Reply:
Hmmm.... i agree that the asymmetry here is somewhat confusing. Thinking in the context of what users have the ability to configure today w/ the `OTEL_METRICS_EXPORTER` [environment variables](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/configuration/sdk-environment-variables.md#exporter-selection) which support `otlp` and `prometheus`:

- `otlp` configures a periodic push based metric reader with OTLP as the exporter
- `prometheus` configures a prometheus metric reader

If we're trying to make the underlying results clear to end users, then the current configuration is "correct", although not really clear. Adding `type: otlp` would be ok if we supported a pull-based OTLP metric reader, but as of now that's not the case. Similarly, if anyone was to add a prometheus remote write exporter, when i would expect the exporter configuration to live under a push-based metric reader.

Reply:
> Adding `type: otlp` would be ok if we supported a pull-based OTLP metric reader, but as of now that's not the case.

I don't understand. Wouldn't the opposite be the case? `otlp` and `prometheus` work as types because they are only pull or push. And new types could of course be added, like `prometheus_remote_write`.

Lining up with the environment variables would also mean configuring the exporter type and not the reader. So maybe like:

```
meter_provider:
  exporters:
    - type: otlp
      interval: 60000
      timeout: 30000
```

```
meter_provider:
  exporters:
    - type: prometheus
      host: "localhost"
      port: 9464
```

Or:

```
meter_provider:
  exporters:
    prometheus:
      host: "localhost"
      port: 9464
```

These seem more inline with what the environment variables are doing. Not that I think this has to stick strictly to the environment variables, but each exporter only works with a specific type of reader, so requiring the user to set that up seems unnecessary and to just add complexity.

Reply:
```
meter_provider:     
  readers:
    - type: prometheus
      host: localhost
      port: 9464
    - type: openmetrics
      host: localhost
      port: 9464
    - type: periodic
       exporter: &otlp
       interval: 60000
```

```
meter_provider:
  readers:
    - type: pull
      exporter:
        type: prometheus
        host: localhost
        port: 9464
    - type: pull
      exporter:
        type: openmetrics
        host: localhost
        port: 9465
    - type: periodic
      exporter: &otlp
      interval: 60000
```

Reply:
I updated the PR to follow the second option shown in @jack-berg's comment above. I think calling out the type of reader explicitly in both push and pull makes it clear to the end user what is happening with less magic.